model/riscv_sys_regs.sail

```diff
@@ -156,21 +156,28 @@ val sys_enable_next = {c: "sys_enable_next", ocaml: "Platform.enable_next", _: "
    unsetting C. If it returns true the write will have no effect. */
 val ext_veto_disable_C : unit -> bool effect {rreg}
 
-/* We currently only support dynamic changes for the C extension. */
 function legalize_misa(m : Misa, v : xlenbits) -> Misa = {
   if   sys_enable_writable_misa ()
-  then { /* Handle modifications to C. */
-         let  v = Mk_Misa(v);
-         /* Suppress changing C if nextPC would become misaligned or an extension vetoes or C was disabled at boot (i.e. not supported). */
-         let m =
-           if   (v.C() == 0b0 & (nextPC[1] == bitone | ext_veto_disable_C())) | not(sys_enable_rvc())
-           then m
-           else update_C(m, v.C());
-         /* Handle updates for F/D. */
-         if   not(sys_enable_fdext()) | (v.D() == 0b1 & v.F() == 0b0)
-         then m
-         else update_D(update_F(m, v.F()), v.D())
-       }
+  then {
+    let v = Mk_Misa(v);
+    /* Suppress disabling C & keep misa unchanged if nextPC would become misaligned or an extension vetoes */
+    if   v.C() == 0b0 & (nextPC[1] == bitone | ext_veto_disable_C())
+    then m
+    else {
+      /* Suppress updates to C if C is read-only */
+      let m = if not(sys_enable_rvc()) then m else update_C(m, v.C());
+      /* Suppress updates to F/D if they are read-only or would become an illegal combination */
```

Comment:
This isn't suppressing F/D updates in the illegal case, it's legalising both to 0, which may or may not be the current value

Reply:
This change is requested in the [comment](https://github.com/riscv/sail-riscv/pull/143#discussion_r839680435) in the original PR. It was not according to the spec before. The privileged spec says about `misa`:

> If an ISA feature x depends on an ISA feature y, then attempting to enable feature x but disable feature y results in both features being disabled. For example, setting “F”=0 and “D”=1 results in both “F” and “D” being cleared.

I have changed the comment, now it is more understandable.